libcloud/container/drivers/rancher.py

```diff
@@ -0,0 +1,1212 @@
+        else:
+            imageuuid = image_type + ':' + image.path
+
+        payload = {
+            "blkioDeviceOptions": blkiodeviceoptions,
+            "build": build,
+            "capAdd": capadd,
+            "capDrop": capdrop,
+            "command": command,
+            "count": count,
+            "cpuSet": cpuset,
+            "cpuShares": cpushares,
+            "dataVolumeMounts": datavolumemounts,
+            "dataVolumes": datavolumes,
+            "dataVolumesFrom": datavolumesfrom,
+            "description": description,
+            "devices": devices,
+            "dns": dns,
+            "dnsSearch": dnssearch,
+            "domainName": domainname,
+            "entryPoint": entrypoint,
+            "environment": environment,
+            "expose": expose,
+            "extraHosts": extrahosts,
+            "healthCheck": healthcheck,
+            "hostname": hostname,
+            "imageUuid": imageuuid,
+            "instanceLinks": instancelinks,
+            "labels": labels,
+            "logConfig": logconfig,
+            "lxcConf": lxcconf,
+            "memory": memory,
+            "memorySwap": memoryswap,
+            "name": name,
+            "networkContainerId": networkcontainerrid,
+            "networkIds": networkids,
+            "networkMode": networkmode,
+            "pidMode": pidmode,
+            "ports": ports,
+            "privileged": privileged,
+            "publishAllPorts": publishallports,
+            "readOnly": readonly,
+            "registryCredentialId": registrycredentialid,
+            "requestedHostId": requestedhostid,
+            "restartPolicy": restartpolicy,
+            "securityOpt": securityopt,
+            "startOnCreate": start,
+            "stdinOpen": stdinopen,
+            "tty": tty,
+            "user": user,
+            "volumeDriver": volumedriver,
+            "workingdir": workingdir
+        }
+
+        return payload
+
+    def ex_list_services(self):
+        """
+        List all Rancher Services
+
+        http://docs.rancher.com/rancher/v1.2/en/api/api-resources/service/
+
+        :rtype: ``list`` of ``dict``
+        """
+
+        result = self.connection.request("%s/services" % self.baseuri).object
+        return result['data']
+
+    def ex_deploy_service(self, name, image, environmentid,
+                          start=True, assignserviceipaddress=None,
+                          service_description=None, externalid=None,
+                          metadata=None, retainip=None, scale=None,
+                          scalepolicy=None, secondarylaunchconfigs=None,
+                          selectorcontainer=None, selectorlink=None,
+                          vip=None, datavolumesfromlaunchconfigs=None,
+                          disks=None, kind=None, memorymb=None,
+                          networklaunchconfig=None, requsetedipaddress=None,
+                          userdata=None, vcpu=None, **kwargs):
+        """
```

Comment:
Not to nitpick too much,  but if we're passing around so many damn arguments, then at least use shorter and pythonic naming conventions to help ease with reading the code, e.g. `networklaunchconfig` -> `net_launch_conf`

I think it would be better to accept a dict of service arguments, e.g. `(self, name, image, env_id, service_args=None, **payload)`